src/ekat/util/ekat_string_utils.hpp

```diff
@@ -61,6 +61,77 @@ std::string join (const Iterable& v, const std::string& sep) {
   return ss.str();
 }
 
+template<typename Iterable, typename Lambda>
+std::string join (const Iterable& v, const Lambda& f, const std::string& sep) {
+  auto it = cbegin(v);
+  auto end = cend(v);
+  if (it==end) {
+    return "";
+  }
+
+  std::stringstream ss;
+  ss << f(*it);
+  ++it;
+  for (; it!=end; ++it) {
+    ss << sep;
+    ss << f(*it);
+  }
+
+  return ss.str();
+}
+
+template<typename Iterable, typename Selector>
+std::string join_if (const Iterable& v, const Selector& check, const std::string& sep) {
+  auto it = cbegin(v);
+  auto end = cend(v);
+  if (it==end) {
+    return "";
+  }
+
+  // It is complicated to figure out if/when to add sep, since we don't
+  // know a priori if the following items will satisfy the check.
+  // Hence, build short list of ptrs beforehand
+  using elem_t = typename std::remove_reference<decltype(*it)>::type;
+  std::list<elem_t*> ptrs;
+
+  for (const auto& elem : v) {
+    if (check(elem)) {
+      ptrs.push_back(&elem);
+    }
+  }
+  auto deref = [](const auto& p) {
+    return *p;
+  };
+  return ekat::join(ptrs,deref,sep);
+}
+
+// Combo of prebious two
```

Comment:
`prebious` -> `previous`

Reply:
whoops, i missed this. hopefully i'll remember to take care of it next time I do a PR.

Reply:
Just make sure you refer to this PR as "a prebious PR" when you do it.